torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3582,14 +3588,13 @@ def issubset(x, y):
             if not issubset(tgt_bound, src_bound):
                 self.log.debug("skipped set_replacement %s = %s (%s) [%s not subset of %s]", a, tgt, msg, tgt_bound, src_bound)
                 return
-            elif (
-                self.is_unbacked_symint(a) and
-                a in self.size_like and
-                free_unbacked_symbols(tgt) and
-                tgt not in self.size_like
-            ):
-                self.log.debug("skipped set_replacement %s = %s (%s) [rhs not size-like]", a, tgt, msg)
-                return
+            elif a in self.size_like:
+                tgt_bound_so = self.bound_sympy(tgt, size_oblivious=True)
+                src_bound_so = self.var_to_range[a] & ValueRanges(2, sympy.oo)
+                if not issubset(tgt_bound_so, src_bound_so):
```

Comment:
I spent a while staring at what this expression meant, which means a comment probably would be helpful.

What you seem to be saying here is that if the target bound doesn't subsume the source bound, we should skip the replacement. This seems fine. But what does it mean for the target bound to subsume the source? You specifically compute what the size-oblivious bound for tgt would be, and compare this against what the size oblivious bound for source would be (in fact, `self.bound_sympy(src, size_oblivious=True)` would be equivalent, and probably clearer!). The reasoning here is that as long as our size-oblivious ranges didn't get worse, it's fine to do this substitution.

This seems correct.

Reply:
So, given the change in the signature of `size_like`, `a` must be a symbol, hence `self.bound_sympy(src, size_oblivious=True)` is the same as `self.var_to_range[a] & ValueRanges(2, sympy.oo)`. Let me add that assert so that's clearer what's going on here.

Will also leave a comment for good measure.